src/start-router.ts

```diff
@@ -7,26 +7,40 @@ import { viewsForDirector } from './utils';
 import { RoutesConfig } from './route';
 import { Store } from './router-store';
 
-const createDirectorRouter = <T extends Store>(views: RoutesConfig<T>, store: T, config = {}) => {
+interface DirectorConfig {
+    html5history?: boolean
+}
+
+const createDirectorRouter = <T extends Store>(views: RoutesConfig<T>, store: T, config: DirectorConfig = {}) => {
     new Router({
         ...viewsForDirector(views, store)
     })
         .configure({
             html5history: true,
             ...config
         })
-        .init();
+        // set fallback to /#/ only when hash routing
+        .init(config.html5history === false ? '/' : undefined);
 };
 
-export const startRouter = <T extends Store>(routes: RoutesConfig<T>, store: T, config = {}) => {
+export const startRouter = <T extends Store>(routes: RoutesConfig<T>, store: T, config: DirectorConfig = {}) => {
     //create director configuration
     createDirectorRouter<T>(routes, store, config);
 
     //autorun and watch for path changes
     autorun(() => {
         const { currentPath } = store.router;
-        if (currentPath !== (window.location.pathname + window.location.search)) {
-            window.history.pushState(null, null || "", currentPath);
+        if (currentPath) {
```

Comment:
`currentPath` can be empty string, and if it's empty,
- pushes empty history (when html5history == true)
- breaks up history with path: `/#/` (when html5history == false)
so I added this condition.